File: Models/PictureSet.cs
using System.Text.Json.Serialization;

namespace KoTi.Models;

public class PictureSet
{
    public int Id { get; set; }

    public string Name { get; set; }
    
    public bool IsPrivate { get; set; }
    
    public int? ParentId { get; set; }
    [JsonIgnore]
    public PictureSet? Parent { get; set; }
    
    public IList<PictureSet> Children { get; set; }
    
    public IList<Picture> Pictures { get; set; }
}

File: KoTi/ClientApp/src/model/PictureSet.ts
export default interface PictureSet {
    id: number;
    name: string;
    isPrivate: boolean;
    parentId: number | null;
    children?: PictureSet[];
    thumbnailUrls: string[];
}

export const BLOG_PICTURES_SET = 'Blog Pictures';
export const COATS_OF_ARMS_SET = 'Coats of Arms';